Fix issue importing bundle

usergenic opened the pull request:
An issue surfaced when trying to execute the shell merge strategy in polymer-build where some imports were not making it into the shell.  I scratched my head a lot and eventually followed the problem to a logic error that came in with mass refactor that had no scenario to cover it.

Anyways, I've added a regression test for the problem scenario (a file linking to the url of its own bundle stopping import of subsequent files) and a fix.

usergenic commented on src/import-utils.ts:
```diff
@@ -67,17 +67,18 @@ export async function inlineHtmlImport(
-    astUtils.removeElementAndNewline(htmlImport);
-    return;
-  }
-
```
So I basically MOVED this block into the if below it, because it is only relevant when the importBundleUrl is not the CURRENT bundle url.

FredKSchott commented on test/html/multiple-imports.html:
```diff
@@ -7,5 +7,6 @@
     Code distributed by Google as part of the polymer project is also
     subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
 -->
+<link rel="import" href="multiple-imports.html">
```
I'm having trouble understanding the issue / when this bug appears. Does a file have to import itself? Isn't that already an error?

usergenic replied:
Fair point.  The test needs to be more intention revealing.  Essentially, bundles with files that are not a part of the original transitive dependency chain will suffer from this problem, since imports are appended in arbitrary order to the hidden div.

In the case of running in polymer-build, what I saw happen was that the common-dependency.html was injected into the shell's bundle by the bundle strategy, because that's what it should do, and shell.html's hidden div wound up with a `<link rel=import href=shell.html><link rel=import href=common-dependency.html>` before inlining, because appendHtmlImports adds references to all files, and then the common-dependency.html wasn't never inlined because the condition that we've "already visited the importBundleUrl of shell.html" short-circuited subsequent imports.

usergenic replied:
This is tricky to represent in a test, but the symptom of the arbitrary injection order was easy enough to replicate by adding a self-reference in the file here.  I'll see if I can clear-up the issue in tests now.

FredKSchott replied:
Ah I think I understand now. Yea, I'd definitely like to see a clean test fixture for this case. New test fixtures are cheap, and I worry about polluting an already-in-use test with something as strange as this.

usergenic replied:
Okay I force pushed a new version of the branch with a test that specifically addresses sharded build strategy (which fails) and subsequent commit contains the fix code (same fix code).

usergenic commented:
@FredKSchott PTAL thx!

FredKSchott commented on src/test/shards_test.ts:
```diff
@@ -97,5 +98,39 @@ suite('Bundler', () => {
+      const strategy = generateShellMergeStrategy(shell, 2);
+      return bundleMultiple([shell, entrypoint1, entrypoint2], strategy)
+          .then((docs) => {
+            //      assert.equal(docs.size, 3);
```
?

usergenic replied:
Good catch- Thank you!  Uncommented.

usergenic commented:
Added a CHANGELOG note too:

```
- Fixed a bug where bundling using generateShellMergeStrategy would result in
  builds missing imports which should have been appended to shell.
```
